src/pb_stub.cc

```diff
@@ -255,6 +255,76 @@ Stub::RunCommand()
     ipc_message = this->PopMessage();
   }
   switch (ipc_message->Command()) {
+    case PYTHONSTUB_CommandType::PYTHONSTUB_AutoCompleteRequest: {
+      // Only run this case when Triton Server is started with
+      // '--strict-model-config=false'
+      bool has_exception = false;
+      std::string error_string;
+      std::string auto_complete_config;
+
+      std::unique_ptr<IPCMessage> auto_complete_response_msg =
+          IPCMessage::Create(shm_pool_, false /* inline_response */);
+      auto_complete_response_msg->Command() = PYTHONSTUB_AutoCompleteResponse;
+      std::unique_ptr<PbString> error_string_shm;
+      std::unique_ptr<PbString> auto_complete_config_shm;
+      AllocatedSharedMemory<AutoCompleteResponseShm> auto_complete_response =
+          shm_pool_->Construct<AutoCompleteResponseShm>();
+
+      ScopedDefer receive_autocomplete_finalize(
+          [this] { stub_message_queue_->Pop(); });
+      ScopedDefer _([this, &auto_complete_response_msg] {
+        SendIPCMessage(auto_complete_response_msg);
+      });
+
+      auto_complete_response.data_->response_has_error = false;
+      auto_complete_response.data_->response_is_error_set = false;
+      auto_complete_response.data_->response_has_model_config = false;
+      auto_complete_response_msg->Args() = auto_complete_response.handle_;
+
+      try {
+        AutoCompleteModelConfig(ipc_message->Args(), &auto_complete_config);
+      }
+      catch (const PythonBackendException& pb_exception) {
+        has_exception = true;
+        error_string = pb_exception.what();
+      }
+      catch (const py::error_already_set& error) {
+        has_exception = true;
+        error_string = error.what();
+      }
+
+      if (has_exception) {
+        // Do not delete the region. The region will be deleted by the parent
+        // process.
+        shm_pool_->SetDeleteRegion(false);
+        LOG_INFO << "Failed to initialize Python stub for auto-complete: "
+                 << error_string;
+        auto_complete_response.data_->response_has_error = true;
+        auto_complete_response.data_->response_is_error_set = false;
+
+        LOG_IF_EXCEPTION(
+            error_string_shm = PbString::Create(shm_pool_, error_string));
+        if (error_string_shm != nullptr) {
+          auto_complete_response.data_->response_is_error_set = true;
+          auto_complete_response.data_->response_error =
+              error_string_shm->ShmHandle();
+        }
+
+        return true;  // Terminate the stub process.
+      }
+
+      if (auto_complete_config != "None" || auto_complete_config.size() != 0) {
```

Comment:
Do we have a test for this? I don't think `auto_complete_config != "None"` works properly. It should be `auto_complete_config == py::none()`.